src/engine/V3Eval.v3

```diff
@@ -981,6 +993,37 @@ convert: u64 -> T, extend: (T) -> Tw, convert_to_u: (Tw) -> Uw) -> u64 {
 	}
 	return res;
 }
+def do_v128_extmul<T, Tw, Uw>(a: (u64, u64), b: (u64, u64), low_half: bool, dest_bits: byte,
+convert: u64 -> T, extend: (T) -> Tw, convert_to_u: (Tw) -> Uw, f: (Tw, Tw) -> Tw) -> (u64, u64) {
+	var src_a = if(low_half, a.0, a.1);	// Choose low or high half of source lanes.
+	var src_high_a = u32.view(src_a >> 32);
+	var src_low_a = u32.view(src_a);
+	var src_b = if(low_half, b.0, b.1);
+	var src_high_b = u32.view(src_b >> 32);
+	var src_low_b = u32.view(src_b);
+	// Extend low and high lanes separately.
+	var dst_low = do_64_extmul_32(src_low_a, src_low_b, dest_bits, convert, extend, convert_to_u, f);
+	var dst_high = do_64_extmul_32(src_high_a, src_high_b, dest_bits, convert, extend, convert_to_u, f);
+	return (dst_low, dst_high);
+}
+def do_64_extmul_32<T, Tw, Uw>(src_a: u32, src_b: u32, dest_bits: byte,
+convert: u64 -> T, extend: (T) -> Tw, convert_to_u: (Tw) -> Uw, f: (Tw, Tw) -> Tw) -> u64 {
+	var res = 0ul;
+	var shift = u6.view(dest_bits >> 1);	// Compute source shift amount, i.e., the width of the original lane.
+	var iters = 64 / dest_bits;	// Compute number of iterations.
+	var res_shift: u6 = 0;
+	var res_shift_inc = u6.view(shift << 1);
+	for (i < iters) {
```

Comment:
You could also do `for (i = 0; i < 64; i += dest_bits)`. That avoids a divide, which is generally an expensive operation.

Reply:
Thanks! Will update it. Similar improvements in other places will be made in another PR.